### PR TITLE
Elastisearch protocol from hardcoded value to config value

### PR DESCRIPTION
**Pull request by jailson-dias**
Currently several of the requests that Vue Storefront is making to elasticsearch are using http protocol fixed direcly in the code, even with the protocol variable in `config/default.json` to be used in the requests. 

So I did this by chenging all requests made to elasticsearch to use the protocol variable from `config/default.json`

This issue was reported in Elastic Search Forces HTTP #108

### CODE REVIEW

File: migrations/.common.js [hunk 1]
```diff
@@ -4,14 +4,20 @@ let kue = require('kue')
 let queue = kue.createQueue(Object.assign(config.kue, { redis: config.redis }))
 
 let es = require('elasticsearch')
+let port = config.elasticsearch.port
+if (!port || port.length < 2) {
+  port = ''
+} else {
+  port = `:${port}`
+}
```
Comment (pkarw) [inline migrations/.common.js]: Wouldn't it be better to check the port with `parseInt()` instead of checking length?

File: scripts/mage2vs.js [hunk 2]
```diff
@@ -21,8 +21,14 @@ function multiStoreConfig(apiConfig, storeCode) {
 
 function getMagentoDefaultConfig(storeCode) {
   const apiConfig = multiStoreConfig(config.magento2.api, storeCode)
+  let port = config.elasticsearch.port
+	if (!port || port.length < 2) {
+		port = ''
+	} else {
+		port = `:${port}`
+	}
```
Comment (pkarw) [inline scripts/mage2vs.js]: The same note as above

File: src/api/catalog.js [hunk 3]
```diff
@@ -62,7 +62,13 @@ export default ({config, db}) => function (req, res, body) {
...
 
 	// pass the request to elasticsearch
-	let url = config.elasticsearch.host + ':' + config.elasticsearch.port + (req.query.request ? _updateQueryStringParameter(req.url, 'request', null) : req.url)
+	let port = config.elasticsearch.port
+	if (!port || port.length < 2) {
+		port = ''
+	} else {
+		port = `:${port}`
+	}
+	let url = `${config.elasticsearch.protocol}://${config.elasticsearch.host}${port}${(req.query.request ? _updateQueryStringParameter(req.url, 'request', null) : req.url)}`
```
Comment (pkarw) [inline src/api/catalog.js]: This snippet is being used here for the third time we need to make it kind of helper please

File: src/api/extensions/gls-parcelshop-dk/index.js [hunk 4]
```diff
@@ -22,7 +22,23 @@ module.exports = ({config, db}) => {
         returnData.droppoints = []
 
         // pass the request to elasticsearch
-        const url = 'http://' + config.elasticsearch.host + ':' + config.elasticsearch.port + '/gls_parcelshop_dk/droppoint/_search';
+        let port = config.elasticsearch.port
+        if (!port || port.length < 2) {
+            port = ''
+        } else {
+            port = `:${port}`
```
Comment (pkarw) [inline src/api/extensions/gls-parcelshop-dk/index.js]: another one :) a shared helper function is required :)

File: src/api/extensions/gls-parcelshop-dk/index.js [hunk 5]
```diff
@@ -22,7 +22,23 @@ module.exports = ({config, db}) => {
...
+        }
+        let url = `${config.elasticsearch.protocol}://${config.elasticsearch.host}${port}/gls_parcelshop_dk/droppoint/_search`
```
Comment (pkarw) [inline src/api/extensions/gls-parcelshop-dk/index.js]: This is something also repeated few times in the code and should be separated to the shared helper function

File: src/graphql/elasticsearch/client.js [hunk 6]
```diff
@@ -1,8 +1,17 @@
...
 import elasticsearch from 'elasticsearch';
 
+
+let port = config.elasticsearch.port
+if (!port || port.length < 2) {
+    port = ''
+} else {
+    port = `:${port}`
+}
+let host = `${config.elasticsearch.protocol}://${config.elasticsearch.host}${port}`
```
Comment (pkarw) [inline src/graphql/elasticsearch/client.js]: Another duplicate

File: src/platform/magento2/tax.js [hunk 7]
```diff
@@ -61,13 +61,20 @@ class TaxProxy extends AbstractTaxProxy {
       inst.applyTierPrices(productList, groupId)
 
       if (this._config.tax.calculateServerSide) {
+        let port = this._config.elasticsearch.port
+        if (!port || port.length < 2) {
+            port = ''
+        } else {
+            port = `:${port}`
+        }
+        let host = `${this._config.elasticsearch.protocol}://${this._config.elasticsearch.host}${port}`
```
Comment (pkarw) [inline src/platform/magento2/tax.js]: To be refactored to shared helper

**Comment by pkarw** [comment]
Plus one more thing - is it possible to add the `ELASTICSEARCH_API_VERSION` variable setup by this change? I mean it should be set to: `config.elasticsearch.apiVersion` in the `mage2vs` script

File: src/api/extensions/gls-parcelshop-dk/index.js (hunk 5 above)
Comment (lukeromanowicz) [reply src/api/extensions/gls-parcelshop-dk/index.js]: Also it should be const instead of let

**Comment by pkarw** [comment]
@jailson-dias i just wanted to kindy ask You about the status and if you're to take the changes required in the CR?

**Pull request closed without merging**